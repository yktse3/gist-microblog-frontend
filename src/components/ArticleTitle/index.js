import React from 'react';
import { string } from 'prop-types';

import {
  Wrapper,
  Title,
  DateTimeString,
} from './styles';

const ArticleTitle = ({ title, dateTime }) => (
  <Wrapper>
    <Title>{title}</Title>
    <DateTimeString>{dateTime}</DateTimeString>
  </Wrapper>
);

ArticleTitle.propTypes = {
  title: string.isRequired,
  dateTime: string.isRequired,
};

export default ArticleTitle;
