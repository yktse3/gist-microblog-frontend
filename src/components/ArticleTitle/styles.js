import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Title = styled.span`
  font-size: 24px;
`;

export const DateTimeString = styled.span`
  font-size: 13px;
`;
