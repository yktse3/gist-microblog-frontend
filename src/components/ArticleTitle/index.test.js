import React from 'react';
import renderer from 'react-test-renderer';
import ArticleTitle from './index';

describe('ArticleTitle', () => {
  it('render successfully', () => {
    const component = renderer.create(
      <ArticleTitle
        title="A title"
        dateTime="2018/10/15 17:13:00"
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
