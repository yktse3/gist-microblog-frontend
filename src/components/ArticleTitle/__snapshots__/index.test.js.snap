// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTitle render successfully 1`] = `
<div
  className="sc-bdVaJa bPGLpq"
>
  <span
    className="sc-bwzfXH dNCrpK"
  >
    A title
  </span>
  <span
    className="sc-htpNat cUROdq"
  >
    2018/10/15 17:13:00
  </span>
</div>
`;
